async/logic/Async4BackendRest_DestinationAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,543 @@
+import static org.apache.kafka.clients.producer.ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.TRANSACTIONAL_ID_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Future;
+
+import javax.ejb.Asynchronous;
+import javax.ejb.Stateless;
+import javax.ws.rs.core.Response;
+
+import com.open.jp.async.errorresponse.ErrorResponse;
+import com.open.jp.async.response.Responses;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.TopicPartition;
+import org.json.JSONObject;
+
+
+@Stateless
+public class Async4BackendRest_DestinationAPI_TopicCheckLogic {
+	
+	private static final String CONSUMER_GROUP_ID = "Async4BackendRest_Group";
+	private static final String OUTPUT_TOPIC = "Async4BackendRest_CallbackAPI_Topic";
+	private static final String INPUT_TOPIC = "Async4BackendRest_DestinationAPI_Topic";
+	
+
+	public Async4BackendRest_DestinationAPI_TopicCheckLogic() {
+		
+	}
+	
+	/**
+	 * Description<br>
+	 * エラー時のレスポンスを生成するメソッド.<br>
+	 * 
+	 * ErrorResponseEnumで定義されている、ErrorStatusCodeMessage、ErrorMessage、ErrorStatusをErrorResponseクラスの
+	 * Time、Code、Messageにセットし各エラーステータスコードによって呼び出す際に利用する.<br>
+	 * 
+	 * @version 2.0
+	 * @param errorresponseenum "E400" "E409" "E00" ErrorResponseEnumで定義されたエラーコードを引数とする.
+	 * @return エラーの場合は"ErrorResponseEnum"クラスで定義されている各エラーとそのメッセージを返す。
+	 */
+	public  Response GeneralErrorRes(ErrorResponseEnum errorresponseenum) {
+	    ErrorResponse erroresponse = new ErrorResponse();
+	    erroresponse.setTime(LocalDateTime.now().toString()); 
+	    erroresponse.setCode(errorresponseenum.getErrorStatusCodeMessage()); 
+	    erroresponse.setMessage(errorresponseenum.getErrorMessage()); 
+	    return Response.status(errorresponseenum.getErrorStatus()).entity(erroresponse).build();
+	}
+	
+	@Asynchronous
+	public Future<Response> ExecThreadAsync4BackendRest_DestinationAPI() {
+
+		KafkaConsumer<String, String> consumer = createKafkaConsumer();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *consumer*. Execution paths that do not contain closure statements were detected. Ensure *consumer* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).